Mathlib/Algebra/Homology/Double.lean

```diff
@@ -182,4 +183,33 @@ noncomputable def evalCompCoyonedaCorepresentableBySingle (i : ι) [DecidableEq
       right_inv f := by simp }
   homEquiv_comp := by simp
 
+variable [c.HasNoLoop] [DecidableEq ι]
+
+open Classical in
+/-- Given a complex shape `c : ComplexShape ι` (with no loop), `X : C` and `j : ι`,
+this is a quite explicit choice of corepresentative of the functor which sends
+`K : HomologicalComplex C c` to `X ⟶ K.X j`. -/
+noncomputable def evalCompCoyonedaCorepresentative (X : C) (j : ι) :
+    HomologicalComplex C c :=
+  if hj : ∃ (k : ι), c.Rel j k ∧ j ≠ k then
```

Comment:
You have probably considered this, but I'll mention that under the `HasNoLoop` assumption this is equivalent to
```suggestion
  if hj : ∃ (k : ι), c.Rel j k then
```
 and then you could derive the `j ≠ h.choose` in the lemma below (probaby most easily with a new lemma `not_rel_of_eq`). But it makes no difference.